docs/encyclopedia/application-message-passing.mdx

```diff
@@ -220,22 +220,39 @@ Every time the Workflow wakes up--generally, it wakes up when it needs to--it wi
 
 This execution is on a single thread–while this means you don’t have to worry about parallelism, you do need to worry about concurrency if you have written Signal and Update handlers that can block. These can run interleaved with the main Workflow and with one another, resulting in potential race conditions. These methods should be made reentrant.
 
+#### Initializing the Workflow first {#workflow-initializers}
+
+Initialize your Workflow's state before handling messages. This prevents reading uninitialized instance variables.
+
+To see why, refer to the [diagram](#message-handler-concurrency). It shows that your Workflow processes messages before the first run of your Workflow's main method.
+
+This happens in two main scenarios:
+
+- When using [Signal-with-Start](#signal-with-start), the message handler always runs first.
+- When your Worker experiences delays, such as during backlogs.
```

Comment:
It would also happen with appreciable probability for any workflow where there is a high rate of signal/update traffic around the time that the workflow is started, e.g. if the workflow ID to be used if communicated to signal/upate callers prior to starting the workflow.